File: 1046-last-stone-weight/1046-last-stone-weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int> maxHeap;
        for(int n:stones){
            maxHeap.push(n);
        }
        while(maxHeap.size()>=2){
            int largest=maxHeap.top();
            maxHeap.pop();

            int second=maxHeap.top();
            maxHeap.pop();
            
            if(largest!=second){
                maxHeap.push(largest-second);
            }
        }
        return maxHeap.empty()?0:maxHeap.top();
    }
};

File: 1046-last-stone-weight/1046-last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        
        stones=[-stone for stone in stones]
        heapq.heapify(stones)
        while len(stones)>=2:
            largest=-heapq.heappop(stones)
            second=-heapq.heappop(stones)
            if largest!=second:
                heapq.heappush(stones, -(largest-second))
        return -stones[0] if stones else 0